contracts/4/BridgedTokenOWL.sol

```diff
@@ -0,0 +1,39 @@
+pragma solidity ^0.4.24;
+
+import "../../vendor/tokenbridge-contracts/contracts/PermittableToken.sol";
+
+/// @dev A contract to implement OWL tokens on a bridged chain (e.g. XDai).
+/// Implements "burning" by allocating the amount to a dedicated receiver,
+/// which can transfer it to the native chain (e.g. Mainnet) and complete
+/// the burn there.
+contract BridgedTokenOwl is PermittableToken {
+    address public burnReceiver;
+
+    event Burnt(address indexed from, address indexed user, uint256 amount);
+
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals,
+        uint256 _chainId,
+        address _burnReceiver
+    ) public PermittableToken(_name, _symbol, _decimals, _chainId) {
+        burnReceiver = _burnReceiver;
+    }
+
+    /// @dev Burns OWL.
+    /// @param user Address of OWL owner
+    /// @param amount Amount of OWL to be burnt
+    function burnOWL(address user, uint256 amount) public {
+        allowed[user][msg.sender] = allowed[user][msg.sender].sub(amount);
```

Comment:
Why not call `transferFrom(user, burnReceiver, amount)` and just add the `Burnt` event? This would actually have the logic that if `user == msg.sender` you don't need to set an allowance.

Reply:
Because then we would have to approve the burnReceiver, rather than the contract that is doing the burn (which is the logic that GP currently relies on).

Reply:
true ... but in this case we always require an allowance by msg.sender, so if `msg.sender == user` this is not convenient. Maybe this is very specific to GP but in this case I would expect an comment here, because this doesn't follow the default burn logic.

Reply:
It follows the logic of the mainnet OWL Token burn:

https://github.com/gnosis/owl-token/blob/602477d11278e66c83cfe275cce71eab1320175d/contracts/5/TokenOWL.sol#L85-L91

I believe the `StandardToken` also requires an allowance even when you call `tansferFrom(address(this), burnReceiver, amount)`. Also keep in mind that if we called `transferFrom` from inside the burnOWL function, then `msg.sender` would actually be the xOWL contract and not user (thus we always need an allowance from user to burnReceiver, whose address might change over time).

Reply:
Since `transferFrom` is a `public` method not `external` the `msg.sender` will not change (as no internal transaction is triggered). 

You are also correct that `StandardToken` is implemented as you mentioned, but the `PermittableToken` overwrites this (https://github.com/poanetwork/tokenbridge-contracts/blob/feb0ba555cb8bf74f7fa21911120d878715a0a84/contracts/PermittableToken.sol#L40)

Reply:
Ah, you are right. When I tried it I must have used `this.transferFrom(user, burnReceiver, amount);` in which case the msg.sender changed.

We could use `transferFrom` then, just that we would have to setAllowance for the burnReceiver which is not convenient as it might change (and you could envision a protocol that does an unlimited approval at instantiation time). We could also drop the require for approval if `msg.sender == user`?

At that point, it would just be to mirror the logic in the original token:

https://github.com/gnosis/owl-token/blob/602477d11278e66c83cfe275cce71eab1320175d/contracts/5/TokenOWL.sol#L86

Reply:
If we have a `setAllowance` won't that emit a `Approval` event and then the xDAI `burnOWL` will be different than the standard one in that it would emit an extra event?

Reply:
Yes it might smarter to stick to the default `burn` behaviour (which is the case with the original implementation) ... I was more focused on the use case that a user wants to call `burnOwl` for his own tokens ... but that is probably also not the most common case.